### ISSUE TITLE
Mod doesn't play nice with Dynamic Trees

### ISSUE THREAD

**Issue by Sluggasaurus** [Ivorforce/RecurrentComplex#349]
For some reason, Recurrent Complex breaks the tree generation in ONLY minecraft's RoofedForest biome.
https://media.discordapp.net/attachments/211957906138988546/486739094592946186/unknown.png?width=956&height=538
Normally, this should just be a forest of dynamic trees, it works with every other biome, but for some reason this is the only one that doesn't work.
Is there an option to disable biome generation with this mod? Is it even an option?

**Comment by IdrisQe** [comment]
I've experienced the same issue. At first I thought it was on Dynamic Trees' end but this is especially bizzare. I even turned Nature generation off in Recurrent Complex's config but Roofed Forests still don't generate Dynamic Trees at all. The weird thing is, the trees that do generate in the Roofed Forest aren't Recurrent Complex trees either, they're just bog-standard Vanilla trees.

**Comment by SleepySukima** [comment]
Same problem, and in case anyone is wondering, yes, disabling Recurrent Complex fixes the issue.
 
I've seen some vanilla trees else where as well, but nowhere near as much as in the Roofed Forest Biome.



I've tried everything. Disabled every tree individually, disabled Nature generation, set it to Lightweight mode, set all the tree and sapling weights to 0, etc.
 
Has anyone managed to solve this?

**Comment by Graywaren** [comment]
Any updates on this issue? Or anyone find a work around? I'd really like to be able to use RC & DT together.  I'm planning to take another whack at figuring it out, but I'm pretty new to mods, so I don't know if I'll be able to identifying anything.

**Comment by Ivorforce** [comment]
I'll be looking into this. 
However, the way the events work unfortunately I can sometimes do little to restore the normal generation state if I do not generate at all. ReC does "emulation" of the biome's forests sometimes if there is no other way to restore it. This might be the cause of vanilla biomes generating rather than Dynamic Trees biomes.

**Comment by Avery93** [comment]
I'm having the same problem despite turning off nature spawning, turning off all associated structures manually in the #structure tweaks GUI, AND blacklisting the roofed forest biome in the config file.

**Comment by Avery93** [comment]
I'm having the same problem despite turning off nature spawning, turning off all associated structures manually in the #structure tweaks GUI, AND blacklisting the roofed forest biome in the config file.

**Comment by naqaden** [comment]
I have Dynamic Trees generating fine after deleting structures from the RC jar itself, such as the **\assets\reccomplex\structures\active\structures\nature\** directory.

**Comment by Avery93** [comment]
I gave that a shot and I'm still getting mostly vanilla trees in roofed forests, which seems to be the only biome that doesn't play nicely with RC when dynamic trees is also installed. Really would have liked for it to work too :(

**Comment by naqaden** [comment]
Now I'm curious! Perhaps it was just my RC config and I overdid it removing items from the jar, or a bit of both is necessary, and/or it's a different mod conflict.

Let's compare configs:
[reccomplex.cfg](https://pastebin.com/C1s44UjK)
[dynamictrees.cfg](https://pastebin.com/dGDLkMLX)

https://text-compare.com/ is a decent solution I just found online for highlighting differences between texts. Javascript is not required. See if anything stands out to you while comparing our RC configs or DT configs.

I just tested again for good measure and there are still no vanilla or RC trees in a new world using the following versions:

MC 1.12.2
Forge 14.23.5.36
IvToolkit 1.3.3
RecurrentComplex 1.4.8.2
DynamicTrees 0.9.5

**Comment by Avery93** [comment]
I actually recently deleted and reset to default all my configs to try and fix an OpenGL error that seemed to have something to do with a configuration conflict, but I hadn''t had any luck fixing roofed forest gen before that either. 

[dynamic trees](https://pastebin.com/bdhd5qeP) & [recurrent complex](https://pastebin.com/ahej8jmC)

MC 1.12.2
Forge 14.23.5.2823
IvToolkit 1.3.3
RecurrentComplex 1.4.8.2
DynamicTrees 0.9.4

I'll try updating DT and Forge, and re-test as well.

**Comment by naqaden** [comment]
The differences that stand out to me are:

My reccomplex.cfg **decorationHacks** is false.
My reccomplex.cfg decoration **baseweights** are 0.0 where trees are concerned, though it sounds like yours were too, previously.
My dynamictrees.cfg **podzolGen** is false. I'll be surprised if that's the culprit, but it does involves worldgen.

**Comment by Avery93** [comment]
Yep, I'd changed  all the weights of nature related gen to 0, and had decoration hacks off. Maybe updating DT will do the trick?

**Comment by naqaden** [comment]
I want to say I didn't have a problem before the latest DT update but it can't hurt to try. I was also on Forge 2807 until recently so I doubt updating that will help either. Maybe it is podzolGen after all? I'm not sure what else to even consider.

If your roofed forests still aren't fully-DT after matching my versions and configs for testing, I'm up for digging into MC and Forge settings.

**Comment by Avery93** [comment]
Matched most of your configs, idk how it could even be podzol gen, since that doesnt happen in roofed forests at all. Also updated forge and dynamic trees, to no effect:
![2019-04-25_23 30 14](https://user-images.githubusercontent.com/48781885/56781788-5fc38700-67b2-11e9-96b0-0bdd5da2fc49.png)

I suppose if it's fine for you, there may be another mod at fault? I've got a few mods that affect world gen, but shouldnt really be affecting the features of individual biomes much, if at all: [BoP](https://minecraft.curseforge.com/projects/biomes-o-plenty), [climatic biome placement](https://minecraft.curseforge.com/projects/climatic-biome-placement), and [geographicraft](https://minecraft.curseforge.com/projects/climate-control-geographicraft). I'll try booting up without those installed and see if anything changes.

**Comment by naqaden** [comment]
I've been testing RC and DT alone for this issue, and until now I'd continued using my edited RC jar without the nature (and other) structures. I just tested a fresh copy of RC and with it I'm able to reproduce the issue despite my config, so gutting the jar is at least part of the solution for sure.

Here's everything I removed from the jar. It's more than I thought, but I didn't need a lot of it because I use **lightweightMode**.

\assets\reccomplex\blockstates
\assets\reccomplex\models
\assets\reccomplex\structures\active\structures\nature
\assets\reccomplex\structures\active\structures\nether
\assets\reccomplex\structures\active\structures\village
\assets\reccomplex\textures

To be clear in case it matters, I didn't just empty those dirs, I deleted the dirs themselves.

**Comment by Avery93** [comment]
I've removed all the same directories (and a few others, namely some of the mazes) so that \assets\reccomplex\ is all but empty, and it still happens. 

![2019-04-26_01 55 38](https://user-images.githubusercontent.com/48781885/56786465-10d41c80-67c7-11e9-8318-11f9a6a068ab.png)

Are the directories the only thing you changed in the modified RC jar? I get the feeling that it's something more fundamental than the structures themselves, but rather something about RC's "emulation" of vanilla biomes as mentioned by Ivorforce previously:

> I'll be looking into this.
> However, the way the events work unfortunately I can sometimes do little to restore the normal generation state if I do not generate at all. ReC does "emulation" of the biome's forests sometimes if there is no other way to restore it. This might be the cause of vanilla biomes generating rather than Dynamic Trees biomes.

Otherwise, I'm pretty much at a loss and resigned to either removing it from regular use or preventing roofed forest generation in the first place, which is a shame since I quite like the biome overall.

**Comment by naqaden** [comment]
The only other thing I modified is the name of the jar... and that appears to be it. I downloaded a fresh copy of RC, prefixed the jar name with a plus sign, kept my existing configs, gen'd a new world... roofed forest was 'dynamic.' Apparently it had nothing to do with removing files.

Now, I feel insane believing in this solution even though it works every time I test it. Can you confirm?

**Comment by Avery93** [comment]
Brilliant! I've also tested and it definitely works. And I've learned another useful thing: I hadn't even realized there was a way to manipulate Forge's load order, which I had thought was a bit of a shortcoming as a mod-management program. Thank you so much!

**Comment by Graywaren** [comment]
Omg fantastic! I will definitely be giving this a shot. Thank you for your awesome detective work.

**Comment by botaxalim** [comment]
Compability with Dynamic Trees mod
just configure the .minecraft\config\reccomplex.cfg

B:generateNature=false


S:baseSaplingSpawnWeight=0

S:baseWeight_cactus=0

S:baseWeight_tree=0

S:saplingTriggerChance=0

then go to .minecraft\mods
rename RecurrentComplex-1.4.8.2.jar to something like +RecurrentComplex-1.4.8.2.jar
you could change "+" with "A" or "B" or "C" example: BRecurrentComplex-1.4.8.2.jar
as long RecurrentComplex on top from DynamicTrees
to compesate load order in mods folder

if you not renaming the jar there will be a problem with Roofed Forest biome trees
or any mods add that biome such as BiomeOPlenty & DynamicTrees & DynamicTreesBOP

**Comment by Avery93** [comment]
@botaxalim yeah... that’s literally what’s already been determined in the whole thread above you

**Comment by botaxalim** [comment]
@Avery93 yeah i know, but i was having hard time, searching in google like 30 min, and the instruction unclear, so i overview and make it easier for other people who search google and end it up here. the steps i provide is very basic even newcomer minecraft modder can do it, well it was write by my and copy pasta on other similar problem topic on curse forge, my intention was helping other if solution works. :D

**Comment by IdrisQe** [comment]
And this, ladies and gentlemen, is why we need load order sorting like Bethesda game modding utilities have. Glad to see someone finally found a workaround for this issue. Can't believe it was as simple as load sorting nameswap.

**Comment by naqaden** [comment]
This could be closed now that [DT v0.9.7](https://www.curseforge.com/minecraft/mc-mods/dynamictrees/files/2870267) forces itself to [load after RC](https://github.com/ferreusveritas/DynamicTrees/commit/c3d3be8a716d9c8e7911acc3b488224766768080) regardless of alphabetical order. Yay!

**Comment by iiTheBeast** [comment]
> 
> 
> Compability with Dynamic Trees mod
> just configure the .minecraft\config\reccomplex.cfg
> 
> B:generateNature=false
> 
> S:baseSaplingSpawnWeight=0
> 
> S:baseWeight_cactus=0
> 
> S:baseWeight_tree=0
> 
> S:saplingTriggerChance=0
> 
> then go to .minecraft\mods
> rename RecurrentComplex-1.4.8.2.jar to something like +RecurrentComplex-1.4.8.2.jar
> you could change "+" with "A" or "B" or "C" example: BRecurrentComplex-1.4.8.2.jar
> as long RecurrentComplex on top from DynamicTrees
> to compesate load order in mods folder
> 
> if you not renaming the jar there will be a problem with Roofed Forest biome trees
> or any mods add that biome such as BiomeOPlenty & DynamicTrees & DynamicTreesBOP

Thanks man I couldnt understand shit given above, probably fixed now.

**Comment by iiTheBeast** [comment]
> 
> 
> This could be closed now that [DT v0.9.7](https://www.curseforge.com/minecraft/mc-mods/dynamictrees/files/2870267) forces itself to [load after RC](https://github.com/ferreusveritas/DynamicTrees/commit/c3d3be8a716d9c8e7911acc3b488224766768080) regardless of alphabetical order. Yay!

I still had a ton of RC trees

**Comment by naqaden** [comment]
> I still had a ton of RC trees

Drat. Thanks for the heads up.
